SCloudRandomizer/MainViewController.m

```diff
@@ -474,33 +335,19 @@ - (void)setupUI:(NSDictionary *)track
     [self.lblArtist setHidden:NO];
     [self.lblLength setHidden:NO];
     
-    long duration = [[track objectForKey:@"duration"] longValue];
-    [self.lblLengthValue setText:[self convertFromMilliseconds:duration]];
-    [self.lblArtistValue setText:[[track objectForKey:@"user"] objectForKey:@"username"]];
-    [self.lblTitleValue setText:[track objectForKey:@"title"]];
-    self.currentTrack = track;
+    [self.lblLengthValue setText:[Utility convertFromMilliseconds:track.duration]];
+    [self.lblArtistValue setText:track.artist];
+    [self.lblTitleValue setText:track.title];
+    
     self.imgArtwork.layer.borderWidth = 1;
     self.imgArtwork.alpha = 1.0;
     self.btnChangeParams.layer.borderColor = [UIColor blackColor].CGColor;
     
-    NSString *urlString = nil;
-    id albumArt = [track objectForKey:@"artwork_url"];
-    if (albumArt == [NSNull null])
-    {
-        urlString = [[[track objectForKey:@"user"] objectForKey:@"avatar_url"] stringByReplacingOccurrencesOfString:@"-large" withString:@"-t300x300"];
-    }
-    else
-    {
-        urlString = [(NSString *)albumArt stringByReplacingOccurrencesOfString:@"-large" withString:@"-t300x300"];
-    }
-    
-    NSURL *imgUrl = [NSURL URLWithString:urlString];
     dispatch_async(dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_BACKGROUND, 0), ^{
```

Comment:
Now that we have a Utils class, we should add a class method that returns a dispatch_get_global_queue. To do this you could just add a new class method to Utils. Something like:

+ (dispatch_queue_t)getGlobalBackgroundQueue;

And in the m file the details would be:

+ (dispatch_queue_t)getGlobalBackgroundQueue
{
    return dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_BACKGROUND, 0);
}